sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -198,17 +221,34 @@ impl Plan {
             .collect::<Vec<String>>();
 
         let mut boundary_ntp_servers = vec![];
+        let mut seen_any_scrimlet = false;
 
         for idx in 0..sled_addrs.len() {
             let sled_address = sled_addrs[idx];
             let subnet: Ipv6Subnet<SLED_PREFIX> =
                 Ipv6Subnet::<SLED_PREFIX>::new(*sled_address.ip());
             let u2_zpools =
                 Self::get_u2_zpools_from_sled(log, sled_address).await?;
-            let mut addr_alloc = AddressBumpAllocator::new(subnet);
+            let is_scrimlet = Self::is_sled_scrimlet(log, sled_address).await?;
 
+            let mut addr_alloc = AddressBumpAllocator::new(subnet);
             let mut request = SledRequest::default();
 
+            // Scrimlets get DNS records for running dendrite
+            if is_scrimlet {
+                let id = Uuid::new_v4();
+                let address = get_switch_zone_address(subnet);
```

Comment:
I'm pretty sure that when creating the service plan with multiple scrimlets, this block of code will add multiple dendrite addresses to the DNS :+1: